### PR TITLE
Update order

### PR DESCRIPTION
**Pull request by Jonathan-M-Wilson**
### Description
- Add spec for update order mutation
- add update method in resolver 
- add update mutation for orderUpdate
- Remove comments from other tests and do test housekeeping

### Testing
- visit code/api/src/modules/order/tests 

### Changes include
- [ ] Bugfix (non-breaking change that solves an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

### Checklist
- [ ] I have written tests for this code (happy & sad)
- [ ] I have updated the Readme
- [x] I ran full test suite - all tests passing

### CODE REVIEW

File: code/api/src/modules/order/tests/orderMutation.test.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  it("order - can update deliveryDate and status", async(done) => {
+    let newDate = 'March 20, 2021 03:24:00'
+    const response = await request(server)
+      .post('/')
+      .send({ query: `mutation { orderUpdate(id: ${order1.id}, deliveryDate: "${newDate}") { deliveryDate status }}`})
+      .expect(200)
+
+      console.log(response.body)
+
+    expect(response.body.data.orderUpdate.deliveryDate).toEqual(new Date(newDate).valueOf().toString())
```
Comment (Todd-Estes) [inline code/api/src/modules/order/tests/orderMutation.test.js:68]: Clutch!

File: code/api/src/modules/order/resolvers.js [hunk 2]
```diff
@@ -27,3 +27,20 @@ export async function getAll() {
...
+  if (auth.isAuthenticated) {
+    await models.Order.update(
+      {
+        deliveryDate,
+        status
+      },
+      {
+        where: { id: id }
+			});
+		return await models.Order.findOne({ where: { id: id } })
```
Comment (helloeduardo) [inline code/api/src/modules/order/resolvers.js:42]: Nice catch here catching the model we were returning and using the id param

File: code/api/src/modules/order/tests/orderMutation.test.js (hunk 1 above)
Comment (helloeduardo) [inline code/api/src/modules/order/tests/orderMutation.test.js:68]: So tricky! But we got it by chaining those methods! 🙌

**Review by Todd-Estes — COMMENTED** [review]
Great work yall!

File: code/api/src/modules/user/tests/userQuery.test.js [hunk 3]
```diff
@@ -50,28 +50,26 @@ describe("user queries", () => {
...
-    // expect(response.body.data.user.subscriptions[0].crate.name).toEqual(crate1.name)
-    // expect(response.body.data.user.subscriptions[0].orders[0].status).toEqual(order1.status)
-    // expect(response.body.data.user.subscriptions[0].orders[0].deliveryDate).toEqual(order1.deliveryDate)
-    // expect(response.body.data.user.subscriptions[0].orders[0].orderProducts[0].returned).toEqual(orderProduct1.returned)
-    // expect(response.body.data.user.subscriptions[0].orders[0].orderProducts[0].product.name).toEqual(product1.name)
+    expect(response.body.data.user.subscriptions[0].crate.name).toEqual(crate1.name)
+    expect(response.body.data.user.subscriptions[0].orders[0].status).toEqual(order1.status)
+    expect(response.body.data.user.subscriptions[0].orders[0].deliveryDate).toEqual(order1.deliveryDate.valueOf().toString())
+    expect(response.body.data.user.subscriptions[0].orders[0].orderProducts[0].returned).toEqual(orderProduct1.returned)
+    expect(response.body.data.user.subscriptions[0].orders[0].orderProducts[0].product.name).toEqual(product1.name)
```
Comment (helloeduardo) [inline code/api/src/modules/user/tests/userQuery.test.js:72]: So glad these tests now pass 🎉

**Review by helloeduardo — APPROVED** [review]
image

**Review by Todd-Estes — APPROVED** [review]

**Review by Todd-Estes — APPROVED** [review]

**Pull request merged**